src/rpc/beacon_api.rs

```diff
@@ -1,27 +1,38 @@
 // Copyright 2019-2024 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
 
-use crate::rpc::error::JsonRpcError;
-use crate::rpc::Ctx;
+use crate::rpc::{
+    reflect::SelfDescribingRpcModule, ApiVersion, Ctx, JsonRpcError, RPCState, RpcMethod,
+    RpcMethodExt as _,
+};
 use crate::{beacon::BeaconEntry, lotus_json::LotusJson, shim::clock::ChainEpoch};
 use anyhow::Result;
 use fvm_ipld_blockstore::Blockstore;
-use jsonrpsee::types::Params;
 
-pub const BEACON_GET_ENTRY: &str = "Filecoin.BeaconGetEntry";
+pub fn register_all(
+    module: &mut SelfDescribingRpcModule<RPCState<impl Blockstore + Send + Sync + 'static>>,
+) {
+    BeaconGetEntry::register(module);
+}
 
 /// `BeaconGetEntry` returns the beacon entry for the given Filecoin epoch. If
 /// the entry has not yet been produced, the call will block until the entry
 /// becomes available
-pub async fn beacon_get_entry<DB: Blockstore>(
-    params: Params<'_>,
-    data: Ctx<DB>,
-) -> Result<LotusJson<BeaconEntry>, JsonRpcError> {
-    let (first,): (ChainEpoch,) = params.parse()?;
-
-    let (_, beacon) = data.beacon.beacon_for_epoch(first)?;
-    let rr =
-        beacon.max_beacon_round_for_epoch(data.state_manager.get_network_version(first), first);
-    let e = beacon.entry(rr).await?;
-    Ok(e.into())
+pub enum BeaconGetEntry {}
+impl RpcMethod<1> for BeaconGetEntry {
+    const NAME: &'static str = "Filecoin.BeaconGetEntry";
```

Comment:
Side quest (likely low priority) - RpcMethod (or something else) should support aliases to make switching between Forest and Lotus easier. https://github.com/filecoin-project/lotus/blob/master/api/eth_aliases.go

Unless it is supported somewhere else already?

Reply:
I think this should be doable in `RpcMethodExt::register` if we add the following member:
```rust
trait RpcMethod {
    const ALIASES: &[&'static str] = [];
}
```

I'll add this to #4032